src/js/shop/components/Checkout/Payment.js

```diff
@@ -0,0 +1,218 @@
+import React, { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import orderBy from "lodash.orderby";
+import { FormattedMessage } from "react-intl";
+import Select from "react-select";
+import useAsync from "react-use/esm/useAsync";
+
+import { PaymentConsumer } from "../../../data/shop/payment";
+import Loader from "../../../components/loaders";
+import { ErrorMessage } from "../Info";
+import { FormField, FormGroup } from "./FormFields";
+import { BodyCart } from "../Cart";
+
+const paymentConsumer = new PaymentConsumer();
+
+const useFetchPaymentMethods = () => {
+    const {
+        loading,
+        error,
+        value = [],
+    } = useAsync(async () => {
+        const methodList = await paymentConsumer.listMethods();
+        return methodList;
+    }, []);
+    const paymentMethods = orderBy(value, ["order"], ["asc"]);
+    return {
+        loading,
+        error,
+        paymentMethods,
+    };
+};
+
+const PaymentMethod = ({
+    id,
+    name,
+    order,
+    logo = null,
+    isSelected = false,
+    onChange,
+}) => {
+    const className = classNames("payment-method", {
+        "payment-method--has-logo": !!logo,
+        "payment-method--active": isSelected,
+    });
+
+    return (
+        <label className={className}>
+            <input
+                type="radio"
+                className="payment-methods__input payment-methods__input--hidde"
+                id={`paymentmethod-${id}`}
+                name="paymentMethod"
+                value={id}
+                onChange={onChange}
+                checked={isSelected}
+            />
+            <div className="payment-method__logo">
+                {logo && <img src={logo} alt={name} />}
+            </div>
+            <span className="payment-method__name">{name}</span>
+        </label>
+    );
+};
+
+PaymentMethod.propTypes = {
+    id: PropTypes.number.isRequired,
+    name: PropTypes.string.isRequired,
+    order: PropTypes.number.isRequired,
+    logo: PropTypes.string,
+    isSelected: PropTypes.bool,
+    onChange: PropTypes.func.isRequired,
+};
+
+const useGetPaymentSpecificOptions = (paymentMethod) => {
+    const {
+        loading,
+        error,
+        value = {},
+    } = useAsync(async () => {
+        if (!paymentMethod) return;
+
+        switch (paymentMethod.name.toLowerCase()) {
+            case "ideal": {
+                const idealBanks = await paymentConsumer.listIdealBanks();
+                return { banks: idealBanks.responseData };
+            }
+        }
+    }, [paymentMethod]);
+
+    if (error) {
+        throw error;
```

Comment:
yeah I'm aware - I'll add a TODO. For now this is intended as a developer-thrown exception, but we should just properly set up error boundaries for these sort of things and throw specific errors that can display a generic but meaningful for the end-user.

> Also not sure how useful this hook really is, seems like the async stuff can be abstracted into a utility function instead that can be used as a callback, otherwise we'd need to return error and loading from here to be used in components.

Not really suitable since it's UI state that is determined by earlier selections, so we can't call these things during the render phase but need a hook for that. And we don't know it upfront since we don't know which payment option the user has selected until they did.

The previous payment options/bank selection flow was broken anyway, since the bank selection is only relevant for ideal payments. It's possible that in the future other payment options will require follow up options that are only relevant for that specific payment option, so this is why it's set up in a "pluggable" way.